### DISCUSSION TITLE
logstash support

### DISCUSSION THREAD

**Discussion by shrynx** [tokio-rs/tracing#2694] [Q&A] [answered]
is there any support for logstash planned ?

or any other alternative ?

**Comment by davidbarsky** [comment, marked as answer]
Sorry for the delay. I've never used Logstash, but I've come across https://github.com/gsson/tracing-logstash with some quick searching. We won't ever really officially support it inside of this repo, but it might be what you're looking for.

**Reply by shrynx** [reply]
i tried it out, i think all it does is provide the log messages in particular format for supporting log stash but i am not sure how do i using tracing library then to send these logs to a specific host like `https://my-etl.my-company.com:6060`

maybe i am just missing some docs here but a quick search didn't lkead me anywhere

**Reply by davidbarsky** [reply]
I don't know enough about logstash, but it seems [its documentation](https://www.elastic.co/guide/en/logstash/current/advanced-pipeline.html#configuring-filebeat) suggests that you use something like Filebeat to collect logs from a file and sent it to your Logstash server.

In my view, having a sidecar daemon be responsible for sending logs is preferable to relying on the application. If your server/application crashes, then the logs written to a file will still be sent, whereas no such guarantee exists with something that directly connects to a server.

**Reply by shrynx** [reply]
that makes a lot more sense. thanks !
